extension/app/ts/background/background.ts

```diff
@@ -494,23 +503,23 @@ async function onContentScriptConnected(port: browser.runtime.Port) {
 				if (request.options.method === 'eth_sendTransaction') return sendTransaction(simulator, port, request, false)
 			}
 
-			return port.postMessage({
+			return postMessageIfStillConnected({
 				interceptorApproved: true,
 				requestId: request.requestId,
 				options: request.options
 			})
 		} catch(error) {
-			console.log('errored!')
-			console.log(error)
-			port.postMessage({
-				interceptorApproved: false,
-				requestId: request.requestId,
-				options: request.options,
-				error: {
-					code: 123456,
-					message: error
-				}
-			})
+			if (error instanceof Error) {
```

Comment:
I feel like we should be propagating the error even if someone decided to throw a non-error.  Consider something like this for ensuring that we are always dealing with an `Error` type (adjusted as necessary to fit this use case).
```ts
error = (error instanceof Error) ? error
	: (typeof error === 'string') ? new Error(error)
	: new Error(JSON.stringify(error))
```